exec/java-exec/src/main/java/org/apache/drill/exec/planner/sql/DirectPlan.java

```diff
@@ -45,11 +52,15 @@ public static PhysicalPlan createDirectPlan(QueryContext context, boolean result
   public static <T> PhysicalPlan createDirectPlan(QueryContext context, T obj){
     Iterator<T> iter = (Iterator<T>) Collections.singleton(obj).iterator();
     return createDirectPlan(context.getCurrentEndpoint(), iter, (Class<T>) obj.getClass());
+  }
 
+  public static <T> PhysicalPlan createDirectPlan(DrillbitEndpoint endpoint, Iterator<T> iterator, Class<T> clazz) {
+    PojoRecordReader<T> reader = new PojoRecordReader<>(clazz, iterator);
+    return createDirectPlan(endpoint, reader, ScanStats.TRIVIAL_TABLE);
   }
-  public static <T> PhysicalPlan createDirectPlan(DrillbitEndpoint endpoint, Iterator<T> iterator, Class<T> clazz){
-    PojoRecordReader<T> reader = new PojoRecordReader<T>(clazz, iterator);
-    DirectGroupScan scan = new DirectGroupScan(reader);
+
+  public static PhysicalPlan createDirectPlan(DrillbitEndpoint endpoint, RecordReader reader, ScanStats stats) {
```

Comment:
Let's keep the default behavior static constructor (which uses trivial) and then we don't have to update the constructor everywhere.